### PR TITLE
Propose Sushi price identifiers

### PR DESCRIPTION
**Pull request by smb2796**

### CODE REVIEW

**Comment by daywiss** [comment]
@smb2796 do you want to tag reviewers on this?

**Comment by smb2796** [comment]
> @smb2796 do you want to tag reviewers on this?

Yep! @mhairimca every Thursday can you tag the reviewers for the week when we know which UMIPs are being reviewed?

File: UMIPs/sushi-price-ids.md [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+| Created    | March 10, 2021   
+| Discourse Link |  
+
+## SUMMARY
+
+The DVM should support price requests for SUSHI/USD, USD/SUSHI, XSUSHI/USD or USD/XSUSHI.
+
+## MOTIVATION
+
+The DVM currently does not support these price identifiers. SUSHI and XSUSHI are also being proposed as supported collateral types, and using these as collateral together with these price identifiers would allow SUSHI or XSUSHI holders to get leverage on their holdings by minting yield-dollar type stablecoins. Adding SUSHI/USD and XSUSHI/USD will also allow for the creation of synthetic Sushi or XSUSHI, or more bespoke contracts such as covered call options backed by SUSHI or XSUSHI.
```
Comment (chicfilabae) [inline UMIPs/sushi-price-ids.md:16]: Are "covered call options" being considered right now? I assume this must be for the folks who are already long SUSHI and want some extra yield by minting this options synth and selling into pool?

File: UMIPs/sushi-price-ids.md [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+- Base Currency: USD
+- Quote currency: XSUSHI
+- Scaling Decimals: 18 (1e18)
+- Rounding: Round to nearest 6 decimal places (seventh decimal place digit >= 5 rounds up and < 5 rounds down)
+
+## RATIONALE
+
+The three markets chosen for pricing SUSHI/USD were picked because they are three of the highest daily trading volume markets available.
+
+SushiSwap has lower daily volume than some other CEXs like OKEx, but it seems correct to use SushiSwap since it is SUSHI's native market and will assumably continue to have one of the highest trading volumes.
```
Comment (chicfilabae) [inline UMIPs/sushi-price-ids.md:89]: Agree.

File: UMIPs/sushi-price-ids.md (hunk 1 above)
Comment (smb2796) [reply UMIPs/sushi-price-ids.md:16]: Yeah what you described would be the primary and most intuitive use case. Covered call options are also being [explored](https://medium.com/uma-project/ulabs-building-call-options-on-uma-part-1-efd3188714c5) though.

File: UMIPs/sushi-price-ids.md [hunk 3]
```diff
@@ -0,0 +1,121 @@
+## Headers
+| UMIP-Sushi    |                                                                                                                                          |
+|------------|------------------------------------------------------------------------------------------------------------------------------------------|
+| UMIP Title | Add SUSHIUSD, USDSUSHI, XSUSHIUSD, USDXSUSHI as supported price identifiers              |
```
Comment (akhanubis) [inline UMIPs/sushi-price-ids.md:4]: Should XSUSHI related ids use / given that they are 9 chars long?

File: UMIPs/sushi-price-ids.md [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+- Is there a cost associated with usage? Yes. Cryptowatch requires a purchase of credits beyond their free tier.
+- If there is a free tier available, how many queries does it allow for?
+
+The free tier is limited to 10 API credits per 24-hours; the cost of querying the market price of a given exchange is 0.005 API credits (i.e. querying the two exchanges will cost 0.010 API credits). This would allow for 1000 free queries per day.
+
+- What would be the cost of sending 15,000 queries? $5.
+
+## PRICE FEED IMPLEMENTATION
+These price identifiers will use price feeds that already exist. Both will use the [Cryptowatch](https://github.com/UMAprotocol/protocol/blob/master/packages/financial-templates-lib/src/price-feed/CryptoWatchPriceFeed.js) and [Uniswap](https://github.com/UMAprotocol/protocol/blob/master/packages/financial-templates-lib/src/price-feed/UniswapPriceFeed.js) price feeds to get the price of SUSHIUSD, and the XSUSHI identifiers will also use the [LPPriceFeed](https://github.com/UMAprotocol/protocol/blob/master/packages/financial-templates-lib/src/price-feed/LPPriceFeed.js) to determine the amount of SUSHI that each XSUSHI is redeemable for.
```
Comment (akhanubis) [inline UMIPs/sushi-price-ids.md:57]: Should MedianizerPriceFeed be included here or is it assumed because of the implementation using median?

**Comment by mhairimca** [comment]
> > @smb2796 do you want to tag reviewers on this?
> 
> Yep! @mhairimca every Thursday can you tag the reviewers for the week when we know which UMIPs are being reviewed?

Didnt realise I should have been doing - will tag from next week on.  Thanks for pointing this out.

File: UMIPs/sushi-price-ids.md [hunk 5]
```diff
@@ -0,0 +1,121 @@
+## Headers
+| UMIP-Sushi    |                                                                                                                                          |
+|------------|------------------------------------------------------------------------------------------------------------------------------------------|
+| UMIP Title | Add SUSHIUSD, USDSUSHI, XSUSHIUSD, USDXSUSHI as supported price identifiers              |
+| Authors    | Sean Brown (smb2796)  |
+| Status     | Draft                                                                                                                                    |
+| Created    | March 10, 2021   
+| Discourse Link |
```
Comment (mhairimca) [inline UMIPs/sushi-price-ids.md:8]: Add discourse link (https://discourse.umaproject.org/t/propose-sushi-price-identifiers/334)

File: UMIPs/sushi-price-ids.md (hunk 4 above)
Comment (mhairimca) [reply UMIPs/sushi-price-ids.md:57]: @daywiss

Comment (smb2796) [reply UMIPs/sushi-price-ids.md:57]: Yes it should. Good point

**Comment by Gamaranto** [comment]
Is this good to go? What needs to be done to get this merged? @mhairimca @smb2796

File: UMIPs/sushi-price-ids.md (hunk 5 above)
Comment (smb2796) [reply UMIPs/sushi-price-ids.md]: Thanks! Added

File: UMIPs/sushi-price-ids.md (hunk 4 above)
Comment (smb2796) [reply UMIPs/sushi-price-ids.md]: Actually used the expression price feed rather than the medianizer.

File: UMIPs/sushi-price-ids.md (hunk 3 above)
Comment (smb2796) [reply UMIPs/sushi-price-ids.md:4]: Perhaps, but I also think that rule is outdated. We discussed a little bit on the community call, and I think I'm going to leave as is.

**Review by Tommy1231232 — APPROVED** [review]

**Pull request merged**